pkg/app/pipedv1/plugin/kubernetes/deployment/annotate_test.go

```diff
@@ -0,0 +1,350 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployment
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/provider"
+)
+
+func TestAnnotateConfigHash(t *testing.T) {
```

Comment:
I almost copied from below and added a testcase for Secrets
https://github.com/pipe-cd/pipecd/blob/e7cbc645c8028d016b40f5f08f559f7343092eb3/pkg/app/piped/executor/kubernetes/kubernetes_test.go#L655